File: sdks/node/src/domain/core/inquirer.ts
import type { CheckboxQ, ConfirmQ, DateQ, PasswordQ, SelectQ, TextQ } from './question.js';

interface IInquirer {
  checkbox(q: CheckboxQ): Promise<string[]>;

  checkbox(q: string, options: string[], id: string, help?: string | null): Promise<string[]>;

  confirm(q: ConfirmQ): Promise<boolean>;

  confirm(q: string, id: string, help?: string | null): Promise<boolean>;

  password(q: PasswordQ): Promise<string>;

  password(q: string, id: string, help?: string | null): Promise<string>;

  select(q: SelectQ): Promise<string>;

  select(q: string, options: string[], id: string, help?: string | null): Promise<string>;

  text(q: TextQ): Promise<string>;

  text(q: string, id: string, help?: string | null): Promise<string>;

  dateSelect(q: DateQ): Promise<string>;

  dateSelect(q: string, id: string, help?: string | null): Promise<string>;
}

export type { IInquirer };

File: sdks/dotnet/sulfone-helium/Domain/Core/Inquirer.cs
using sulfone_helium.Domain.Core.Questions;

namespace sulfone_helium.Domain.Core;

public interface IInquirer
{
    Task<string[]> Checkbox(CheckboxQ q);
    Task<string[]> Checkbox(string q, string[] options, string id, string? help = null);
    Task<bool> Confirm(ConfirmQ q);
    Task<bool> Confirm(string q, string id, string? help = null);
    Task<string> Password(PasswordQ q);
    Task<string> Password(string q, string id, string? help = null);
    Task<string> Select(SelectQ q);
    Task<string> Select(string q, string[] options, string id, string? help = null);
    Task<string> Text(TextQ q);
    Task<string> Text(string q, string id, string? help = null);
    Task<DateOnly> DateSelect(DateQ q);
    Task<DateOnly> DateSelect(string question, string id, string? help = null);
}
